### ISSUE TITLE
Option to ignore file if unsupported shebang

### ISSUE THREAD

**Issue by OJFord** [koalaman/shellcheck#2425]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I use shellcheck primarily with pre-commit; this works well for files I can find by pattern `*.sh` or whatever.

With executable scripts that don't have a file extension though, the only way I could glob them with pre-commit to pass to shellcheck would be as say not containing `.` - but then this may well include other files (`Dockerfile` comes to mind) that are not shell scripts.

I think a good solution to this would be if shellcheck had a flag to ignore the file (not error) if the file doesn't have (or has an unsupported) shebang line.

**Comment by guillaume-d** [comment]
Indeed, I have exactly the same problem while using shellcheck over [sds/overcommit](https://github.com/sds/overcommit/). There one might try to get clever with inclusion/exclusion patterns, but I find @OJFord's proposed solution better.

**Comment by OJFord** [comment]
For the pre-commit case, I stumbled across [its ability to filter files by 'type'](https://pre-commit.com/index.html#filtering-files-with-types) instead of filename regex, and specifically the relatively recent addition of `types_or` to combine them in disjunction.

This means I can use:
```yaml
- id: shell-lint
  args:
    - -x
  types_or:
    - bash
    - sh
    - shell
```

and `pre-commit` takes care of finding anything that ends `.sh`, `.bash` (the `shell` element), or has `sh` or `bash` shebang lines (the first two elements).

Pretty nice, maybe `overcommit` has something similar (or would be interested in adding it)? I'll leave this open though, since although that pretty much solves it for me, it is more generally applicable and could be nice just to run `shellcheck` over an entire directory (possibly not even under SVC) and pick everything up.
